scripts/pre-commit.git-lint.sh

```diff
@@ -14,8 +14,13 @@
 # limitations under the License.
 
 # First part return the files being commited, excluding deleted files.
-git diff-index -z --cached HEAD --name-only --diff-filter=ACMRTUXB |
-xargs --null --no-run-if-empty git lint;
+DIFF=$(git diff-index -z --cached HEAD --name-only --diff-filter=ACMRTUXB)
+if [ -z "$DIFF" ]
+then
+  exit 0;
+else
+  git lint;
```

Comment:
Note that this is not exactly the same as the original script. The original script was linting only files being commited. 

Here it should read something like:
```
echo $DIFF_FILES | xargs -0 git lint
```

The same applies for the mercurial hook.